File: 백준/Silver/6550. 부분 문자열/부분 문자열.py
import sys
input = sys.stdin.readline

while True:
    text = input().strip()
    if not text:
        break
    s,t = text.split()
    idx = 0
    for char in t:
        if idx<len(s) and char == s[idx]:
            idx +=1
    print("Yes" if idx == len(s) else "No")

File: 백준/Silver/6550. 부분 문자열/부분 문자열.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String line;
        while((line=br.readLine())!= null && !line.trim().isEmpty()){
            String[] input = line.trim().split(" ");
            String s = input[0];
            String t = input[1];
            int idx = 0;
            for(char c:t.toCharArray()){
                if(idx<s.length() && c==s.charAt(idx)){
                    idx ++;
                }
            }
            System.out.println(idx==s.length()?"Yes":"No");
        }
    }
}